File: core/src/main/java/com/ustadmobile/core/catalog/contenttype/ScormTypePlugin.java
package com.ustadmobile.core.catalog.contenttype;

import com.ustadmobile.core.view.ScormPackageView;

import org.xmlpull.v1.XmlPullParserException;

import java.io.IOException;
import java.io.InputStream;
import java.util.Arrays;
import java.util.List;

/**
 * Created by mike on 1/6/18.
 */

public class ScormTypePlugin extends ContentTypePlugin {

    public static final String[] MIME_TYPES = new String[]{"application/scorm+zip"};

    @Override
    public String getViewName() {
        return ScormPackageView.VIEW_NAME;
    }

    @Override
    public List<String> getMimeTypes() {
        return Arrays.asList(MIME_TYPES);
    }

    @Override
    public List<String> getFileExtensions() {
        return Arrays.asList(new String[]{"zip"});
    }

}

File: core/src/main/java/com/ustadmobile/core/catalog/contenttype/ScormTypePlugin.kt
package com.ustadmobile.core.catalog.contenttype

import com.ustadmobile.core.view.ScormPackageView

import org.xmlpull.v1.XmlPullParserException

import java.io.IOException
import java.io.InputStream
import java.util.Arrays

/**
 * Created by mike on 1/6/18.
 */

class ScormTypePlugin : ContentTypePlugin() {

    override val viewName: String
        get() = ScormPackageView.VIEW_NAME

    override val mimeTypes: List<String>
        get() = Arrays.asList(*MIME_TYPES)

    override val fileExtensions: List<String>
        get() = Arrays.asList(*arrayOf("zip"))

    companion object {

        val MIME_TYPES = arrayOf("application/scorm+zip")
    }

}
